controllers/jenkins/pipelinerun/pipelinerun_controller.go

```diff
@@ -165,47 +165,94 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 			log.Error(err, "unable to update PipelineRun status.")
 			return ctrl.Result{RequeueAfter: time.Second}, err
 		}
-		r.recorder.Eventf(&pr, corev1.EventTypeNormal, v1alpha3.Updated, "Updated running data for PipelineRun %s", req.NamespacedName)
+		r.recorder.Eventf(pipelineRunCopied, corev1.EventTypeNormal, v1alpha3.Updated, "Updated running data for PipelineRun %s", req.NamespacedName)
 		// until the status is okay
 		// TODO make the RequeueAfter configurable
 		return ctrl.Result{RequeueAfter: 3 * time.Second}, nil
 	}
 
 	// first run
-	pipelineBuild, err := r.triggerJenkinsJob(namespaceName, pipelineName, &pr.Spec)
+	jobRun, err := r.triggerJenkinsJob(namespaceName, pipelineName, &pipelineRunCopied.Spec)
 	if err != nil {
 		log.Error(err, "unable to run pipeline", "namespace", namespaceName, "pipeline", pipeline.Name)
-		r.recorder.Eventf(&pr, corev1.EventTypeWarning, v1alpha3.TriggerFailed, "Failed to trigger PipelineRun %s, and error was %s", req.NamespacedName, err)
+		r.recorder.Eventf(pipelineRunCopied, corev1.EventTypeWarning, v1alpha3.TriggerFailed, "Failed to trigger PipelineRun %s, and error was %s", req.NamespacedName, err)
 		return ctrl.Result{}, err
 	}
-	log.Info("Triggered a PipelineRun", "runID", pipelineBuild.ID)
+	// check if there is still a same pending PipelineRun
+	exists, err := r.hasSamePendingPipelineRun(jobRun, pipeline)
+	if err != nil {
+		log.Error(err, "unable to check if there still has the same pending PipelineRun", "jobRun", jobRun)
+		return ctrl.Result{}, err
+	}
+	if exists {
+		// if there still exists the same pending PipelineRun, then give up reconciling
+		if err := r.deletePipelineRun(req.NamespacedName); err != nil {
+			log.Error(err, "unable to delete PipelineRun")
+			return ctrl.Result{}, err
```

Comment:
what happened if the pipelinerun was deleted. At the same time, you will print the log `unable to delete PipelineRun`